org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/utils/LogHandler.java

```diff
@@ -0,0 +1,58 @@
+package org.eclipse.lsp4xml.utils;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.logging.Handler;
+import java.util.logging.LogRecord;
+import java.util.logging.SimpleFormatter;
+
+import org.eclipse.lsp4j.MessageParams;
+import org.eclipse.lsp4j.MessageType;
+import org.eclipse.lsp4xml.XMLLanguageServer;
+
+/**
+ * LogHandler
+ */
+public class LogHandler extends Handler{
+  private XMLLanguageServer languageServer;
+
+  public LogHandler(XMLLanguageServer languageServer) {
+    this.languageServer = languageServer;
+  }
+
+  @Override
+  public void publish(LogRecord record) {
+    if(languageServer == null) {
+      return;
+    }
+    StringBuilder sb = new StringBuilder();
+    sb.append(record.getMillis())
+    .append(" - ")
+    .append(record.getSourceClassName())
+    .append("#")
+    .append(record.getSourceMethodName())
+    .append(" - ")
+    .append(record.getMessage());
+    if (record.getThrown() != null) {
+      StringWriter sw = new StringWriter();
+      PrintWriter pw = new PrintWriter(sw);
+      pw.println();
+      record.getThrown().printStackTrace(pw);
+      pw.close();
+      String throwable = sw.toString();
+      sb.append(throwable);
+    }
```

Comment:
all of the above should be moved to a `String format(LogRecord record)` method, for testing purposes